profiler/src/ProfilerEngine/Datadog.Profiler.Native/LibddprofExporter.cpp

```diff
@@ -249,15 +249,22 @@ ddog_Endpoint LibddprofExporter::CreateEndpoint(IConfiguration* configuration)
         // Agent mode
 
 #if _WINDOWS
-        bool useDefaultDomainSocket = false;
+        std::string socketPath;
+        const std::string& namePipeName = configuration->GetNamedPipeName();
+        if (!namePipeName.empty())
+        {
+            socketPath = R"(\\.\pipe\)" + namePipeName;
+        }
+        std::string agentUrl = "windows:" + socketPath;
 #else
-        std::error_code ec; // fs::exists might throw if no error_code parameter is provided
-        bool useDefaultDomainSocket = fs::exists("/var/run/datadog/apm.socket", ec);
+        std::string socketPath = "/var/run/datadog/apm.socket";
+        std::string agentUrl = "unix://" + socketPath;
 #endif
 
-        if (useDefaultDomainSocket)
+        std::error_code ec; // fs::exists might throw if no error_code parameter is provided
+        if (fs::exists(socketPath, ec))
         {
-            _agentUrl = "unix:///var/run/datadog/apm.socket";
+            _agentUrl = agentUrl;
         }
         else
         {
```

Comment:
Note that customers can also select UDS (in tracer) by setting `DD_APM_RECEIVER_SOCKET` - this is actually an _agent_ variable, though we look for it too (if the agent is using it, it's the one we need). I believe this is _generally_ undocumented though, so not sure if we want to do that (@pierotibou has Thoughts)

Reply:
I think Lucas was very against that and we had actually left it in the code because it had been added and it was a breaking change to remove it. And [we haven't documented internally either](https://github.com/DataDog/architecture/blob/master/rfcs/apm/integrations/trace-autodetect-agent-config/rfc.md#traces), so I assume we shouldn't add it even though it could be a pain for the few users that would have used it.

Reply:
Oh, that's right. IMO, it's kind of strange to _not_ use it (especially as it's back compat). Otherwise it's like we're saying "Ok, we _know_ where the agent is listening, but you _have_ to tell us explicitly". There _are_ reasons to not use it (conflicting values etc) but I'm not sure they're _good_ reasons 🤔

Reply:
So, if someone that has already setup the tracer uses this variable, it would be good for them to have CP OOTB I believe. So yes I think it'd be better to add it. But in that case, should we also handle legacy stuff as agentPort and agentHost? 
That said, I assume it's outside the scope of this PR.

Reply:
Re: `DD_APM_RECEIVER_SOCKET`

In retrospect, it does make sense for the library to use it if it happens to be set for the agent, but mainly as a fallback if the library transport is not otherwise configured.